src/schoology-extractor/tests/test_schoology_request_service.py

```diff
@@ -0,0 +1,367 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+from logging import Logger
+from typing import Tuple
+
+import pandas as pd
+import pytest
+from unittest.mock import Mock
+
+from schoology_extractor.schoology_request_service import SchoologyRequestService
+from schoology_extractor.api.request_client import RequestClient
+from schoology_extractor.api.paginated_result import PaginatedResult
+from schoology_extractor.mapping import users as usersMap
+
+
+def describe_when_getting_users():
+    def describe_given_one_user():
+        @pytest.fixture
+        def result() -> pd.DataFrame:
+            logger = Mock(spec=Logger)
+            request_client = Mock(spec=RequestClient)
+            page_size = 22
+
+            users = {
+                "user": [{"uid": 1234, "role_id": 321}],
+                "total": 1,
+                "links": {"self": "ignore"},
+            }
+            users_page = PaginatedResult(
+                request_client, page_size, users, "user", "ignore me"
+            )
+
+            roles = {"role": [{"id": 321, "title": "estudiante"}]}
+            roles_page = PaginatedResult(
+                request_client, page_size, roles, "role", "ignore me"
+            )
+
+            # Arrange
+            request_client.get_users.return_value = users_page
+            request_client.get_roles.return_value = roles_page

```

Comment:
Here's where I was getting tripped up yesterday morning: I thought that I needed `request_client.return_value`. That would have been true if I had been using `mock.patch`, but in this case `request_client` _is_ the mock object already.